docs/method-monitoring.asciidoc

```diff
@@ -0,0 +1,282 @@
+Of course, the higher the level of detail, the higher the profiler overhead and the Elasticsearch index sizes.
+As most of the processing is done in the background, the impact on the response time of user requests is negligible.
+
+**Clean up clutter in the APM app**
+
+Filter out inferred spans that are faster than the configured threshold,
+and avoid cluttering the APM app with fast-executing methods by setting
+<<config-profiling-inferred-spans-min-duration,`profiling_inferred_spans_min_duration`>>.
+
+**Include and exclude specific classes**
+
+Include and exclude specific classes explicitly with
+<<config-profiling-inferred-spans-included-classes,`profiling_inferred_spans_included_classes`>> and
+<<config-profiling-inferred-spans-excluded-classes,`profiling_inferred_spans_excluded_classes`>>.
+Generally, the fewer classes that are included, the faster and the more memory efficient the processing is.
+
+By default, the classes from the JDK and from most application servers are excluded. This reduces the number of uninteresting inferred spans.
+
+**Example `elasticapm.properties` file with inferred spans enabled**
+
+[source,properties]
+----
+profiling_inferred_spans_enabled=true
+profiling_inferred_spans_sampling_interval=50ms
+profiling_inferred_spans_min_duration=25ms
+profiling_inferred_spans_included_classes=org.example.myapp.*
+profiling_inferred_spans_excluded_classes=(?-i)java.*
+----
+
+[float]
+==== Caveats
+
+Inferred spans are estimations, not exact measurements.
+They may start after the method actually started, and end before the method actually ended.
+This can lead to inconsistencies, all of which are documented in the
+https://github.com/elastic/apm-agent-java/tree/master/apm-agent-plugins/apm-profiling-plugin[apm-profiling-plugin readme].
+
+// Inferred spans are created after the profiling session ends and therefor may appear later in the APM app's span timeline than regular spans.
+// Because of this, a regular span cannot be the child of an inferred span.
+
+//********************************************
+[[method-api]]
+=== API/Code
+
+Use the <<api-span,span API>> to manually create spans for methods of interest.
+The API is extremely flexible, and offers the ability to customize your spans,
+by adding labels to them, or by changing the type, name, or timestamp.
+
+TIP: OpenTracing fan? You can use the <<opentracing-bridge,OpenTracing API>>, instead of the Agent API,
+to manually create spans.
+
+[float]
+==== How to create spans with the span API
+
+. Get the current span with <<api-current-span,`currentSpan()`>>,
+which may or may not have been created with auto-instrumentation.
+. Create a child span with <<api-span-start-span,`startSpan()`>>.
+. Customize the span with the <<api-span,span API>>.
+
+[source,java]
+----
+import co.elastic.apm.api.ElasticApm;
+import co.elastic.apm.api.Span;
+
+Span parent = ElasticApm.currentSpan(); <1>
+Span span = parent.startSpan(); <2>
```

Comment:
Activating the span is important. See also https://github.com/elastic/apm-agent-java/issues/1127

Reply:
Good to know. I added a callout to the code example to point this out.